src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -127,3 +189,8 @@ function showHidePipelineSection(link) {
         }
     }
 }
+
+window.addEventListener('load', function() {
+    isParallelBuild($$('.pipeline-new-node'));
+    generatePipelineLinks($$('.pipeline-new-node'));
```

Comment:
Doing this only once on page load seems like it will have issues if someone opens the log for an in-progress build, leaves it open while the build continues, and then comes back and looks at the logs. For example, there may not be a parallel step when the page is opened initially, but one may execute later on. I think that in-progress builds are supported today by `Behaviour.specify` and the old code, but I haven't specifically tested it.

Reply:
I will test this use case with the current code and will let you know. It is also a variant to remove the parallel check all-together - the performance impact will be minimal, as the main performance issue was fixed.